changelog/7594.feature.rst

```diff
@@ -0,0 +1 @@
+Enable support of subclassing :class:`~sunpy.coordinates.frames.SunPyBaseCoordinateFrame` outside the core library.
```

Comment:
I think this should mention the fact it's `wcs_to_coordinate_frame` or whatever the astropy api is.

Reply:
Yea it's the `celestial_frame_to_wcs` and then `wcs_to_celestial_frame` the other way. So what something like

```
Enable support of subclassing :class:`~sunpy.coordinates.frames.SunPyBaseCoordinateFrame` outside the core library utilising :func:`astropy.wcs.utils.wcs_to_celestial_frame`.

Reply:
I would reorder that sentence a little and say "ignore unknown subclasses of SunPyBaseCoordinateFrame by default in wcs_to_celestial_frame"?